src/backend/libinput/mod.rs

```diff
@@ -191,11 +193,21 @@ impl backend::PointerAxisEvent<LibinputInputBackend> for PointerScrollAxis {
         }
     }
 
-    fn source(&self) -> backend::AxisSource {
+    fn source(&self) -> AxisSource {
         match self {
-            Self::Wheel(_) => backend::AxisSource::Wheel,
-            Self::Finger(_) => backend::AxisSource::Finger,
-            Self::Continuous(_) => backend::AxisSource::Continuous,
+            Self::Wheel(_) => AxisSource::Wheel,
+            Self::Finger(_) => AxisSource::Finger,
+            Self::Continuous(_) => AxisSource::Continuous,
+        }
+    }
+
+    fn relative_direction(&self, axis: Axis) -> backend::AxisRelativeDirection {
+        let device = backend::Event::<LibinputInputBackend>::device(self);
+        match axis {
+            Axis::Vertical if device.config_scroll_natural_scroll_enabled() => {
```

Comment:
Does it only apply to the vertical axis? The explanation of natural scrolling, "matching the motion of the scroll device with the direction of the content", does not read like it should be restricted to the vertical axis.

Reply:
Seems you're right, when I test the natural scrolling option in libinput for either a touchpad or mouse.

Reply:
The Kwin implementation in https://invent.kde.org/plasma/kwin/-/merge_requests/3456 has a comment `Maps to wl_pointer.axis_relative_direction. Currently used for y axis only`, but I'm not sure that's true in any was as implemented there...